frontend/lib/util/strings.ts

```diff
@@ -10,5 +10,5 @@ export function ellipsis(text: string, maxLength: number = 20): string {
 // returns undefined when non-numeric characters are encountered
 export function parseIntStrict(text: string): number | undefined {
   const num = parseInt(text, 10);
-  return !isNaN(num) && num.toString() === text ? num : undefined;
```

Comment:
We should add a new function with a new name for this, instead of changing the behaviour of `parseIntStrict` to also strip zeroes.